### PR TITLE
Meta information for VSNetBeans 12.6.301

### PR DESCRIPTION
**Pull request by JaroslavTulach**
Preparing release of VSCode NetBeans extension 12.6.301 from branch [vsnetbeans_preview_1263](https://github.com/apache/netbeans/tree/vsnetbeans_preview_1263).

### CODE REVIEW

**Review by neilcsmith-net — CHANGES_REQUESTED** [review]
Please create a duplicate file for VSNetBeans adjacent to this. Ideally set the default URL to it in the VSNetBeans build scripts too.

Originally adding here / doing separate releases was going to be a one off. Ongoing it's problematic using the same file.

**Comment by JaroslavTulach** [comment]
I've noticed Eric's request in #45, however I am afraid separate file (`vsnetbeans.json`) will not work well for regular (e.g. `12.5.0`, `12.6.0`, etc.) releases. As such I decided to propose the _traditional_ way of handling the release meta-information.

**Comment by neilcsmith-net** [comment]
That fixes the major but not only issue. Separate file will work fine for in sync releases too, as long as we copy across correct branch info.

Let's wait on @ebarboni review.

File: meta/netbeansrelease.json [hunk 1]
```diff
@@ -586,6 +586,35 @@
...
     },
+    "release1263": {
+        "position": "16",
+        "ant": "ant_latest",
+        "jdk": "jdk_1.8_latest",
+        "jdk_apidoc": "https://docs.oracle.com/javase/8/docs/api/",
+        "maven": "maven_3.3.9",
+        "versionName": "12.6.301",
+        "tlp": "true",
+        "apidocurl": "https://bits.netbeans.org/12.6/javadoc",
```
Comment (JaroslavTulach) [inline meta/netbeansrelease.json:597]: I've just started [run 807](https://ci-builds.apache.org/job/Netbeans/job/netbeans-vscode/807/) with [parameter METAURL](https://ci-builds.apache.org/job/Netbeans/job/netbeans-vscode/807/parameters/) set to `https://raw.githubusercontent.com/apache/netbeans-jenkins-lib/0f4c351e89e1945ae440f53d0ac7c2ba9c0acae4/meta/netbeansrelease.json`. According to the [log output](https://ci-builds.apache.org/job/Netbeans/job/netbeans-vscode/807/consoleFull) the branding info was picked properly up:
```
[releasejson] Writing releasinfo file /home/jenkins/jenkins-agent/workspace/Netbeans/netbeans-vscode/nbbuild/build/netbeansrelease.xml
[releasejson] Branding computed info: metabuild.DistributionURL=https://netbeans.apache.org/nb/updates/12.6/updates.xml.gz?{$netbeans.hash.code}
[releasejson] Branding computed info: metabuild.PluginPortalURL=https://netbeans.apache.org/nb/plugins/12.6/catalog.xml.gz
[releasejson] Branding computed info: metabuild.RawVersion=12.6.301
[releasejson] Branding computed info: metabuild.apachepreviousversion=9.0,10,11.0,11.1,11.2,11.3,12.0,12.1,12.2,12.3,12.4,12.5,12.5.301,12.6
[releasejson] Branding computed info: metabuild.ComputedSplashVersion=12.6.301
[releasejson] Branding computed info: metabuild.ComputedTitleVersion=12.6.301
[releasejson] Branding computed info: metabuild.logcli=
```

Comment (ebarboni) [reply meta/netbeansrelease.json:597]: Would it work with the following PR https://github.com/apache/netbeans/pull/3443 applied ? I did'nt remember the blocker on pre 12.4 vscode. If it pass it should not break the upcoming vscode release.

**Comment by neilcsmith-net** [comment]
@JaroslavTulach please don't edit my comments. The problem is that `netbeansrelease.json` is not just setting up the information for the current version, it was intended to be a canonical list of previous _IDE_ releases. It's caused hours of work already due to breaking that fact.  Anything that needs to pick up a list of previous releases gets broken, not just the Javadoc generation.  At some point we might need to add more logic in to filter by what is and isn't a NetBeans IDE release. If that's build side, then VSNetBeans will still have to behave differently.

TL;DR I agree about one point of truth, but the truth I'd like to preserve is an accurate snapshot of release history.

File: meta/netbeansrelease.json (hunk 1 above)
Comment (neilcsmith-net) [reply meta/netbeansrelease.json:597]: @JaroslavTulach yes, it's expected to be right for VSNetBeans. Although you can see that `metabuild.apachepreviousversion` is incorrect both here and in NetBeans IDE. Not a major concern at the moment as it doesn't break anything, but it demonstrates the inaccurate picture of release history.

**Comment by neilcsmith-net** [comment]
@ebarboni with `netbeansrelease.properties` in the source releases now, I assume builds from source releases for VSNetBeans will not be affected by later changes to `netbeansrelease.json`?  If that's the case, we could allow this in now to allow the release to proceed, then resolve where this information is kept longer term before we build releases of NetBeans 13?

**Comment by ebarboni** [comment]
lets merge I guess

**Comment by neilcsmith-net** [comment]
@ebarboni as long as that keeps the _possibility_ of moving into a separate file when we branch for 13 next week, I'm good with it.

**Review by neilcsmith-net — APPROVED** [review]

**Comment by neilcsmith-net** [comment]
@ebarboni just realised from something in the VSCode vote thread that we don't have the option to split, because the JSON file is still being parsed in source bundle builds, even though the results are _correctly_ discarded!  Need to fix this by 13 release.  I'll look at moving the JSON download and parsing into an optional task based on the properties file existing.  Is the JSON file copied into the build folder for any other reason?
